scrapers/lobbyist/scrape_filings.py

```diff
@@ -1,79 +1,139 @@
+from abc import ABC, abstractmethod
+
 import csv
 import json
 import logging
 import scrapelib
 import sys
+import click
 
 from tqdm import tqdm
 
 
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.DEBUG)
 
-reader = csv.DictReader(sys.stdin)
-writer = csv.DictWriter(
-    sys.stdout,
-    fieldnames=[
-        "ClientID",
-        "ReportName",
-        "ReportFileName",
-        "ReportTypeCode",
-        "ReportType",
-        "PeriodStart",
-        "PeriodEnd",
-        "DueDate",
-        "SubmittedDate",
-        "TotalRows",
-        "ReportID",
-        "ReportVersionID",
-        "Status",
-        "IncidentalLobbying",
-        "ReportStatus",
-        "IsAfterDueDate",
-        "CreatedBy",
-        "CreatedDate",
-        "CreatedIPAddress",
-        "LobbyistClientID",
-        "LobbyistClientVersionID",
-        "LastModifiedBy",
-        "LastModifiedDate",
-        "LastModifiedIPAddress",
-        "Message",
-        "IsSuccess",
-        "Action",
-        "IsDirty",
-        "PersonID",
-        "PersonVersionID",
-        "MemberID",
-    ],
-)
-
-writer.writeheader()
-
-s = scrapelib.Scraper(requests_per_minute=0, retry_attempts=3, verify=False)
-
-payload = {
-    "PageNo": 1,
-    "PageSize": 1000,
-    "SortDir": "ASC",
-    "SortedBy": "",
-    "ElectionYear": "",
-    "LobbyistVersionID": "",
-}
-
-for row in tqdm(reader):
-    _payload = payload.copy()
-    _payload["LobbyistID"] = row["MemberID"]
-    _payload["LobbyistVersionID"] = row["MemberVersionID"]
-
-    response = s.post(
-        "https://login.cfis.sos.state.nm.us/api//ExploreClients/Fillings",
-        data=json.dumps(_payload),
-        headers={"Content-Type": "application/json"},
+
+class LobbyistScraper(ABC, scrapelib.Scraper):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    @abstractmethod
+    def _get_payload(self, id, version):
+        pass
+
+    def scrape(self, url, id, version):
+        payload = self._get_payload(id, version)
+
+        response = self.post(
+            url,
+            data=json.dumps(payload),
+            headers={"Content-Type": "application/json"},
+            verify=False,
+        )
+
+        if response.ok:
+            for record in response.json():
+                # Add the lobbyist ID and version to the filing record
+                record["MemberID"] = id
+                return record
+
+
+class LobbyistEmployerScraper(LobbyistScraper):
+    url = "https://login.cfis.sos.state.nm.us/api//ExploreClients/Disclosures"
+
+    def _get_payload(self, id, version):
+        return {
+            "PageNumber": 1,
+            "PageSize": 1000,
+            "SortDir": "ASC",
+            "SortedBy": "",
+            "Year": "",
+            "ClientID": id,
+            "ClientVersionID": version,
+        }
+
+    def scrape(self, id, version):
```

Comment:
Why is this call signature different from the abstract base class?

Reply:
I couldn't figure out a clean way for the `LobbyistScraper` base class to reference the `url` attribute in a child class. 

[Turns out you can't stack @classmethod and @property to define a static class attribute as of 3.11.](https://docs.python.org/3/whatsnew/3.11.html#language-builtins)

If I'm overlooking something, please let me know.

Reply:
How about stacking @property and @abstractmethod? https://docs.python.org/3/library/abc.html#abc.abstractproperty

Reply:
Good idea!